addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -40,6 +42,28 @@ options.UserValueWidget.include({
     },
 });
 
+Many2oneUserValueWidget.include({
+    /**
+     * @override
+     */
+    async _getSearchDomain() {
+        // Add the current website's domain if the model has a website_id field.
+        // Note that the `_rpc` method is cached in Many2X user value widget,
+        // see `_rpcCache`.
```

Comment:
I've [adapted](https://github.com/odoo/odoo/compare/448087354369f9fa1a9dfc4cf9929a88dd70939b..f8bb635347fab389c577f83ff55f3b33d0f3e193) this comment, let's not add a `TODO` that we won't probably ever do and that we probably don't want, it was a possibility of implementation in the spec that I made but since the `_rpc` is already cached this is ok for now.
(Actually the cache is to be adapted tho, as it's caching the value and not the promise which doesn't prevent multiple simultaneous call to fire the same RPC).